stock_quant_package_dimension/models/stock_quant_package.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class StockQuantPackage(models.Model):
+    _inherit = 'stock.quant.package'
+
+    weight = fields.Float('Weight (kg)')
+    length = fields.Integer('Length (mm)', help='length in millimeters')
+    width = fields.Integer('Width (mm)', help='width in millimeters')
+    height = fields.Integer('Height (mm)', help='height in millimeters')
+    volume = fields.Float(
+        'Volume (m³)',
+        digits=(8, 4),
+        compute='_compute_volume',
+        readonly=True,
+        store=False,
+        help='volume in cubic meters',
+    )
+
+    @api.depends('length', 'width', 'height')
+    def _compute_volume(self):
+        for pack in self:
+            pack.volume = (
+                pack.length * pack.width * pack.height
+            ) / 1000.0 ** 3
+
+    def auto_assign_packaging(self):
+        self = self.with_context(_auto_assign_packaging=True)
+        res = super().auto_assign_packaging()
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super().write(vals)
+        if (
+            self.env.context.get('_auto_assign_packaging')
+            and vals.get('product_packaging_id')
+        ):
+            for package in self:
+                package.onchange_product_packaging_id()
+        return res
+
+    @api.onchange('product_packaging_id')
+    def onchange_product_packaging_id(self):
+        # FIXME somehow the values returned by this onchange are not handled
+        #  correctly in basic_model.js on _.each at line 1594 where an overflow
+        #  is happening: https://github.com/odoo/odoo/issues/41353
+        if self.product_packaging_id:
+            vals = self.product_packaging_id.read(
+                fields=['length', 'width', 'height', 'weight']
```

Comment:
```suggestion
                fields=['length', 'width', 'height', 'weight', 'max_weight']
```

Reply:
or maybe not... But you are not reading that field but using it on line 54. And then you are reading weight, but overwriting the value with max_weight, so it is strange.